### ISSUE TITLE
Ctrlora implementation : Easy solution to train a controlnet with < 1000 examples

### ISSUE THREAD

**Issue by mamad-sd** [huggingface/diffusers#9713]
### Model/Pipeline/Scheduler description

Authors of the paper trained a base controlnet (with a new architecture if I'm not mistaken) on 9 different conditions to allow finetuning on new conditions easily using a LoRA rank 128. This method allows finetuning a novel condition using less than 1000 examples with less than 24GB of VRAM in a few hours. 
The potential is really high and for having trained a new unknown condition, I can confirm it works pretty well (even though my dataset has 5K examples but I did train it quickly using only a 3090 GPU).

The only problem is that the training and inference code seems to be done on the old stable diffusion code and it may be difficult to port it to diffusers.

Anyone interested in implementing the training and inference code in diffusers ? License is Apache-2.0 license

### Open source status

- [X] The model implementation is available.
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Ctrlora repo : https://github.com/xyfJASON/ctrlora
Ctrlora paper : https://arxiv.org/abs/2410.09400

**Comment by sayakpaul** [comment]
I _think_ this could just be about:

1. Initialize `ControlNetModel` from the pretrained UNet with the `from_unet()` method. 
2. Call `add_adapter()` on the model with the respective LoRA params. For this `ControlNetModel` needs to also subclass from `PeftLoaderMixin`. 
3. Train the model. 

The pre-trained LoRAs are here:
https://huggingface.co/xyfJASON/ctrlora/tree/main/ctrlora-loras/base-conditions

So, for inference and support loading these, 

1. We convert the LoRA state dict into a form that can be loaded into a `ControlNetModel` just like we do for the UNet: https://github.com/huggingface/diffusers/blob/9dcac8305749de1eea84dcc53367bfac9b2bc35b/src/diffusers/loaders/unet.py#L354
2. Now, this ControlNetModel can be used with the pipeline we have for ControlNets. 

Would you maybe like to give this a try? Happy to guide. 

Cc: @asomoza

**Comment by jloveric** [comment]
Is this being implemented as part of this PR? https://github.com/huggingface/diffusers/pull/10686
